nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs

```diff
@@ -151,6 +154,22 @@ public void UpdateWeapon(Sprite sprite)
                 _clonedSkin.SetAttachment(_weaponSlotIndex, WeaponSlot, newWeapon);
             }
 
+            Destroy(_cachedWeaponVFX);
+
+            if (!(weaponVFXPrefab is null))
+            {
+                var parent = new GameObject(weaponId.ToString());
+                parent.AddComponent<BoneFollower>();
+                parent.transform.SetParent(transform);
```

Comment:
Is it ok to add component on every update of weapon?

Reply:
The weapon effect cached before attaching the component is being destroyed.